c2pa/c2pa_api/c2pa_api.py

```diff
@@ -114,6 +114,10 @@ def add_ingredient(self, ingredient, format, stream):
 
     def add_ingredient_file(self, ingredient, path):
         format = os.path.splitext(path)[1][1:]
+        if "title" not in ingredient:
+            if isinstance(ingredient, str):
+                ingredient = json.loads(ingredient)
```

Comment:
Should this be wrapped in try/catch in case the json loading fails?

Reply:
Possibly, but caller should be catching exceptions like this. I'll do an updated pass on all the errors and exceptions when I review the main SDK errors.